database/sql/src/main/scala/cromwell/database/slick/tables/MetadataEntryComponent.scala

```diff
@@ -181,44 +186,24 @@ trait MetadataEntryComponent {
-    * keys are returned.
-    */
-  def metadataEntriesNotLikeMetadataKeys(workflowExecutionUuid: String, metadataKeys: NonEmptyList[String],
-                                         requireEmptyJobKey: Boolean) = {
-    (for {
-      metadataEntry <- metadataEntries
-      if metadataEntry.workflowExecutionUuid === workflowExecutionUuid
-      if !metadataEntryHasMetadataKeysLike(metadataEntry, metadataKeys)
-      if metadataEntryHasEmptyJobKey(metadataEntry, requireEmptyJobKey)
-    } yield metadataEntry).sortBy(_.metadataTimestamp)
-  }
+  private[this] def metadataEntryHasMetadataKeysLike(metadataEntry: MetadataEntries,
+                                                     metadataKeysToFilterFor: List[String],
+                                                     metadataKeysToFilterOut: List[String]): Rep[Boolean] = {
 
-  /**
-    * Returns metadata entries that are NOT "like" metadataKeys for the specified call.
-    * If jobIndex (resp. jobAttempt) has no value, all metadata keys for all indices (resp. attempt)
-    * are returned.
-    */
-  def metadataEntriesNotLikeMetadataKeysWithJob(workflowExecutionUuid: String, metadataKeys: NonEmptyList[String],
-                                                callFqn: String, jobIndex: Option[Int], jobAttempt: Option[Int]) = {
-    (for {
-      metadataEntry <- metadataEntries
-      if metadataEntry.workflowExecutionUuid === workflowExecutionUuid
-      if !metadataEntryHasMetadataKeysLike(metadataEntry, metadataKeys)
-      if metadataEntry.callFullyQualifiedName === callFqn
-      if hasSameIndex(metadataEntry, jobIndex)
-        // Assume that every metadata entry for a call should have a non null attempt value
-        // Because of that, if the jobAttempt parameter is Some(_), make sure it matches, otherwise take all entries
-        // regardless of the attempt
-      if (metadataEntry.jobAttempt === jobAttempt) || jobAttempt.isEmpty
-    } yield metadataEntry).sortBy(_.metadataTimestamp)
-  }
+    def containsKey(key: String): Rep[Boolean] = metadataEntry.metadataKey like key
 
-  private[this] def metadataEntryHasMetadataKeysLike(metadataEntry: MetadataEntries,
-                                                     metadataKeys: NonEmptyList[String]): Rep[Boolean] = {
-    metadataKeys.toList.map(metadataEntry.metadataKey like _).reduce(_ || _)
+    val positiveFilter: Option[Rep[Boolean]] = metadataKeysToFilterFor.map(containsKey).reduceOption(_ || _)
+    val negativeFilter: Option[Rep[Boolean]] = metadataKeysToFilterOut.map(containsKey).reduceOption(_ || _)
+
+    (positiveFilter, negativeFilter) match {
+      case (Some(pf), Some(nf)) => pf && !nf
+      case (Some(pf), None) => pf
+      case (None, Some(nf)) => !nf
+
+      // We should never get here, but there's no reason not to handle it:
+      // ps: is there a better literal "true" in slick?
+      case (None, None) => metadataEntry.metadataEntryId === metadataEntry.metadataEntryId
```

Comment:
IIRC you can

```scala
  val include: Rep[Boolean] = true
```

Reply:
Exactly what ☝️ said. One may be able to use a literal Rep[Boolean] with something like:

```scala
case (None, None) => true: Rep[Boolean]
```